applications/crossbar/src/modules/cb_modules_util.erl

```diff
@@ -666,29 +665,26 @@ group_by_assign_to([{DID, Assign}|NumUpdates], Groups) ->
 -spec format_assignment_results(knm_numbers:ret()) -> assignment_updates().
 format_assignment_results(#{ok := OKs
                            ,ko := KOs}) ->
-    format_assignment_oks(OKs) ++ format_assignment_kos(maps:to_list(KOs)).
+    format_assignment_oks(OKs) ++ format_assignment_kos(KOs).
 
 -spec format_assignment_oks(knm_number:knm_numbers()) -> assignment_updates().
 format_assignment_oks(Numbers) ->
-    lists:foldl(fun(Number, Acc) ->
-                        PN = knm_number:phone_number(Number),
-                        [{knm_phone_number:number(PN), {'ok', Number}} | Acc]
-                end, [], Numbers).
+    [{knm_phone_number:number(PN), {'ok', Number}}
+     || Number <- Numbers,
+        PN <- [knm_number:phone_number(Number)]
+    ].
 
--spec format_assignment_kos([{knm_numbers:num(), knm_numbers:ko()}]) -> assignment_updates().
--spec format_assignment_kos([{knm_numbers:num(), knm_numbers:ko()}], assignment_updates()) ->
-                                   assignment_updates().
+-spec format_assignment_kos(knm_numbers:kos()) -> assignment_updates().
 format_assignment_kos(KOs) ->
-    format_assignment_kos(KOs, []).
+    maps:fold(fun format_assignment_kos_fold/3, [], KOs).
 
-format_assignment_kos([], Acc) -> Acc;
-format_assignment_kos([{Num, Reason}|KOs], Acc) when is_atom(Reason) ->
-    Acc1 = [{Num, {'error', Reason}} | Acc],
-    format_assignment_kos(KOs, Acc1);
-format_assignment_kos([{Num, ReasonJObj}|KOs], Acc) ->
+-spec format_assignment_kos_fold(knm_numbers:num(), knm_numbers:ko(), assignment_updates()) ->
+                                        assignment_updates().
+format_assignment_kos_fold(Number, Reason, Updates) when is_atom(Reason) ->
+    [{Number, {'error', Reason}} | Updates];
+format_assignment_kos_fold(Number, ReasonJObj, Updates) ->
     Error = knm_errors:error(ReasonJObj),
```

Comment:
I think you don't need to extract the `error` here, as your `ReasonJObj` was already the thing returned in `{error, ReasonJObj}` before the knm_numbers rewrite. Just returning the `error` actually changes the way the HTTP API works.